### PR TITLE
Add more breadcrumbs to better understand Attachments issues

### PR DESCRIPTION
**Pull request by KevinBoulongne**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt [hunk 1]
```diff
@@ -104,11 +104,43 @@ object SentryDebug {
...
     }
 
+    fun addAttachmentsBreadcrumb(draft: Draft) = with(draft) {
+
+        var count = 1
+        val data = mutableMapOf<String, Any>()
+
+        fun String.keyPad(): String = padStart(length = 15)
+        fun Int.countPad(): String = toString().padStart(length = 2, '0')
+        fun count(): String = "${count.countPad().also { count++ }}."
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:114]: `Pad` is a bit weird. I had to search what the padStart method did to understand what was the meaning of all this.

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt [hunk 2]
```diff
@@ -104,11 +104,43 @@ object SentryDebug {
...
+        fun format(index: Int): String = (index + 1).countPad()
+
+        data[count() + "email".keyPad()] = AccountUtils.currentMailboxEmail.toString()
+
+        data[count() + "draft".keyPad() + " - localUuid"] = localUuid
+        data[count() + "draft".keyPad() + " - remoteUuid"] = remoteUuid.toString()
+        data[count() + "draft".keyPad() + " - action"] = action?.name.toString()
+        data[count() + "draft".keyPad() + " - mode"] = when {
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:122]: You could create an intermediate variable with `"count() + "draft".keyPad() + " -"`

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt [hunk 3]
```diff
@@ -104,11 +104,43 @@ object SentryDebug {
...
+            forwardedUid != null -> "FORWARD"
+            else -> "NEW_MAIL"
+        }
+
+        data[count() + "attachments".keyPad() + " - count"] = attachments.count()
+
+        attachments.forEachIndexed { index, it ->
+            data[count() + "attachment #${format(index)}".keyPad()] =
+                "localUuid: ${it.localUuid} | uuid: ${it.uuid} | uploadLocalUri: ${it.uploadLocalUri}"
+            data[count() + "attachment #${format(index)}".keyPad()] = "uploadStatus: ${it.uploadStatus.name} | size: ${it.size}"
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:133]: Also here with `count() + "attachment #${format(index)}".keyPad()`

**Review by tevincent — CHANGES_REQUESTED** [review]

**Comment by tevincent** [comment]
You also have a code smell.

**Review by LunarX — CHANGES_REQUESTED** [review]
Add a keyword so each different step where we record a breadcrumb can be identified so when analyzing the breadcrumbs we know which step introduced the issue

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt (hunk 2 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:123]: I think it complicates the readability of the code.

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt (hunk 3 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:134]: I think it complicates the readability of the code.

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt (hunk 1 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:114]: What do you propose ?

Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:114]: Fixed in #1988.

File: app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt (hunk 3 above)
Comment (KevinBoulongne) [reply app/src/main/java/com/infomaniak/mail/utils/SentryDebug.kt:134]: Fixed in #1988.

**Pull request merged**